pkg/jmxfetch/limiter.go

```diff
@@ -0,0 +1,42 @@
+package jmxfetch
+
+import "time"
+
+// restartLimiter checks if too many restarts happened within a time period
+type restartLimiter struct {
+	maxRestarts int
+	interval    float64
+	stopTimes   []time.Time
+	idx         int
+}
+
+func newRestartLimiter(maxRestarts int, interval float64) restartLimiter {
+	var stopTimes []time.Time
+	if maxRestarts > 0 {
+		stopTimes = make([]time.Time, maxRestarts)
+	}
+	return restartLimiter{
+		maxRestarts: maxRestarts,
+		interval:    interval,
+		stopTimes:   stopTimes,
+		idx:         0,
+	}
+}
+
+func (r *restartLimiter) canRestart(now time.Time) bool {
+	if r.maxRestarts < 1 {
+		return false
+	}
+
+	r.stopTimes[r.idx] = now
+	oldestIdx := (r.idx + 1) % r.maxRestarts
```

Comment:
This used to be `oldestIdx := (idx + maxRestarts + 1) % maxRestarts` - did you intend to drop `maxRestarts` from the calculation?

Reply:
Yes, as far as I can tell, adding `maxRestarts` to the lhs is redundant in this case (all values are non-negative here). Do you think we should keep it anyway?

Reply:
Makes sense - just wanted to double check. 
in that case line `39` below can just be: `r.idx = oldestIdx` so you don't have to do it twice :)